controllers/grid/settings/user/form/UserDetailsForm.php

```diff
@@ -265,6 +301,12 @@ public function execute(...$functionParams)
             $this->user->setInlineHelp(1); // default new users to having inline help visible
         }
 
+        // if doing only a partial update that includes only updating user's user group
+        if ($this->userGroupUpdateOnly) {
+            parent::execute(...$functionParams);
```

Comment:
I think because there are now security implications to how `execute` functions, I'd propose rearranging the implementation in this and related classes (`UserForm` and `UserRoleForm`, which also extends `UserForm`). Otherwise it might be possible that some change to `UserForm` accidentally permits users of `UserDetailsForm` to modify more or less of the user object than they should be able to.

I'd recommend renaming `UserForm::execute` to something like `saveUserGroupAssignments`, and then having both `UserDetailsForm` and `UserRoleFarm` call both `$this->saveUserGroupAssignments` **and** `parent::execute` (which will have become the `Form::execute` implementation, just for the sake of hooks).

(I don't like the way the `Form` class functions that take care of hook calls have had implications that ripple through subclass implementations -- but that can be cleaned up another time!)

Reply:
done.